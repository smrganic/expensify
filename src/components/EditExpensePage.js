import React from "react"
import { connect } from "react-redux"
import ExpenseForm from "./ExpenseForm"
import { editExpense, removeExpense } from "../redux/actions/expenses"

export class EditExpensePage extends React.Component {

    onSubmit = (expense) => {
        this.props.editExpense(this.props.expense.id, expense)
        this.props.history.push("/")
    }

    onClick = (event) => {
        this.props.removeExpense(this.props.expense.id)
        this.props.history.push("/")
    }

    render() {
        return (
            <div>
                <ExpenseForm
                    expense={this.props.expense}
                    onSubmit={this.onSubmit}
                />
                <button onClick={this.onClick}>Remove</button>
            </div>
        )
    }
}

// Implicitly returns an object
const mapStateToProps = (state, props) => ({
    expense: state.expenses.find((expense) => expense.id === props.match.params.id)
})

const mapDispatchToProps = (dispatch) => ({
    editExpense: (id, expense) => dispatch(editExpense(id, expense)),
    removeExpense: (id) => dispatch(removeExpense({ id }))
})

export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage)