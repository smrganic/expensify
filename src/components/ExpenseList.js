import React from "react"
import ExpenseItem from "./ExpenseListItem"
import { connect } from "react-redux"
import selectExpenses from "../redux/selectors/expenses"

const ExpenseList = (props) => (
    <div>
        <h1>ExpenseList</h1>
        {props.expenses.map((expense) => (
            <ExpenseItem key={expense.id} {...expense} />
        ))}
    </div>
)

const mapStateToProps = (state) => ({
    expenses: selectExpenses(state.expenses, state.filters)
})

const ConnectedExpenseList = connect(mapStateToProps)(ExpenseList)

export default ConnectedExpenseList