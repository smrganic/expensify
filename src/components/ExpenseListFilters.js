import React from "react"
import { connect } from "react-redux"
import { DateRangePicker } from "react-dates"
import { setTextFilter, sortByDate, sortByAmount, setStartDate, setEndDate } from "../redux/actions/filters"
import moment from "moment"

export class ExpenseListFilters extends React.Component {

    state = {
        calendarFocused: null
    }

    onDatesChange = ({ startDate, endDate }) => {
        this.props.setStartDate(startDate)
        this.props.setEndDate(endDate)
    }

    onFocusChange = (calendarFocused) => {
        this.setState(() => ({ calendarFocused }))
    }

    onTextChange = (event) => {
        this.props.setTextFilter(event.target.value)
    }

    onSortChange = (event) => {
        if (event.target.value === "date") {
            this.props.sortByDate()
        } else {
            this.props.sortByAmount()
        }
    }

    render() {
        return (
            <div>
                <input
                    txpe="text"
                    value={this.props.filters.text}
                    onChange={this.onTextChange}
                />
                <select
                    value={this.props.filters.sortBy}
                    onChange={this.onSortChange}>
                    <option value="date">Date</option>
                    <option value="amount">Amount</option>
                </select>
                <DateRangePicker
                    startDate={this.props.filters.startDate}
                    startDateId={moment().toString()}
                    endDate={this.props.filters.endDate}
                    endDateId={moment().toString()}
                    onDatesChange={this.onDatesChange}
                    focusedInput={this.state.calendarFocused}
                    onFocusChange={this.onFocusChange}
                    showClearDates={true}
                    numberOfMonths={1}
                    isOutsideRange={() => false}
                />
            </div>
        )
    }
}

const mapStateToProps = (state) => ({ filters: state.filters })

const mapDispatchToProps = (dispatch) => ({
    setTextFilter: (text) => dispatch(setTextFilter(text)),
    sortByAmount: () => dispatch(sortByAmount()),
    sortByDate: () => dispatch(sortByDate()),
    setStartDate: (startDate) => dispatch(setStartDate(startDate)),
    setEndDate: (endDate) => dispatch(setEndDate(endDate))
})

export default connect(mapStateToProps, mapDispatchToProps)(ExpenseListFilters)