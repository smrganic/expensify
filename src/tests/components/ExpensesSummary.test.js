import React from "react"
import { shallow } from "enzyme"
import { ExpensesSummary } from "../../components/ExpensesSummary"

test("Should render ExpensesSummary with one expense.", () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={255} />)
    expect(wrapper).toMatchSnapshot()
})

test("Should render ExpensesSummary with multiple expenses.", () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={50} expensesTotal={123454321} />)
    expect(wrapper).toMatchSnapshot()
})