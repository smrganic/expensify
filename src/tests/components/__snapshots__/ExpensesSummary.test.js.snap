// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render ExpensesSummary with multiple expenses. 1`] = `
<div>
  <h1>
    Viewing 
    50
     
    expenses
     totalling 
    $1,234,543.21
  </h1>
</div>
`;

exports[`Should render ExpensesSummary with one expense. 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totalling 
    $2.55
  </h1>
</div>
`;
